server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/UserMappingRoutesTest.java

```diff
@@ -0,0 +1,229 @@
+                "  {" +
+                "    \"type\": \"Group\"," +
+                "    \"mapping\": \"group123@domain.tld\"" +
+                "  }" +
+                "]");
+    }
+
+    @Test
+    void getUserMappingsShouldReturnSameMappingsWhenParametersInUpperCase() throws RecipientRewriteTableException {
+        recipientRewriteTable.addAddressMapping(
+            MappingSource.fromUser(User.fromUsername("alice123@domain.tld")), "user123@domain.tld");
+        recipientRewriteTable.addAliasMapping(
+            MappingSource.fromUser(User.fromUsername("alice123@domain.tld")), "aliasuser123@domain.tld");
+        recipientRewriteTable.addGroupMapping(
+            MappingSource.fromUser(User.fromUsername("alice123@domain.tld")), "group123@domain.tld");
+
+        recipientRewriteTable.addAddressMapping(
+            MappingSource.fromUser(User.fromUsername("bob456@domain.tld")), "user456@domain.tld");
+        recipientRewriteTable.addAliasMapping(
+            MappingSource.fromUser(User.fromUsername("bob456@domain.tld")), "aliasuser456@domain.tld");
+        recipientRewriteTable.addGroupMapping(
+            MappingSource.fromUser(User.fromUsername("bob456@domain.tld")), "group456@domain.tld");
+
+        String jsonBody = when()
+                .get("AliCE123@domain.tld")
+            .then()
+                .contentType(ContentType.JSON)
+                .statusCode(HttpStatus.OK_200)
+            .extract()
+                .body()
+                .asString();
+
+        assertThatJson(jsonBody)
+            .when(Option.IGNORING_ARRAY_ORDER)
+            .isEqualTo("[" +
+                "  {" +
+                "    \"type\": \"Address\"," +
+                "    \"mapping\": \"user123@domain.tld\"" +
+                "  }," +
+                "  {" +
+                "    \"type\": \"Alias\"," +
+                "    \"mapping\": \"aliasuser123@domain.tld\"" +
+                "  }," +
+                "  {" +
+                "    \"type\": \"Group\"," +
+                "    \"mapping\": \"group123@domain.tld\"" +
+                "  }" +
+                "]");
+    }
+
+    @Test
+    void getUserMappingShouldReturnBadRequestWhenInvalidParameter() throws RecipientRewriteTableException {
+        recipientRewriteTable.addAddressMapping(
+            MappingSource.fromUser(User.fromUsername("alice123@domain.tld")), "user123@domain.tld");
+        recipientRewriteTable.addAliasMapping(
+            MappingSource.fromUser(User.fromUsername("alice123@domain.tld")), "aliasuser123@domain.tld");
+        recipientRewriteTable.addGroupMapping(
+            MappingSource.fromUser(User.fromUsername("alice123@domain.tld")), "group123@domain.tld");
+
+        when()
+            .get("alice123@domain@domain.tld")
+        .then()
+            .statusCode(HttpStatus.BAD_REQUEST_400);
+    }
+
+    @Test
+    void getUserMappingShouldReturnServerErrorWhenMappingSourceIsFromDomain() throws RecipientRewriteTableException {
```

Comment:
the method name says it should return server error(500) but in the code you assert it to be ok(200). By the way I think this scenario is not really related to the route, I prefer to remove this test.